### PR TITLE
Define Path type

### PR DESCRIPTION
**Pull request by kaloyan-raev**
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/101)

### CODE REVIEW

File: examples/paths/main.go [hunk 1]
```diff
@@ -30,33 +29,36 @@ func main() {
 
 // Main is the exported CLI executable function
 func Main() error {
-	fmt.Printf("mnemonic: %s\n", mnemonic)
+	fmt.Println("mnemonic:", mnemonic)
```
Comment (nfarah86) [inline examples/paths/main.go]: should we replace Println with something else? logger.Debug ? or use the zap library?

File: pkg/paths/paths.go [hunk 2]
```diff
@@ -9,41 +9,72 @@ import (
...
+
+// Encrypt creates new Path by encrypting the current path with the given key
+func (p Path) Encrypt(key []byte) (encrypted Path, err error) {
+	encrypted = make([]string, len(p))
+	for i, seg := range p {
+		encrypted[i], err = encrypt(seg, key)
 		if err != nil {
 			return nil, err
 		}
 		key = deriveSecret(key, seg)
```
Comment (nfarah86) [inline pkg/paths/paths.go]: may be I totally missed this, but what do you do with key?

**Comment by kaloyan-raev** [comment]
Review status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @kaloyan-raev)

---

*[examples/paths/main.go, line 32 at r1](https://reviewable.io/reviews/storj/storj/101#-LFT15Cat3UiU4RnVB9T-r1-32:-LFq1DrE1xywmPGvzGUV:b-wkaqfq) ([raw file](https://github.com/storj/storj/blob/74e6203d1e3bd701e53b299a4f9d98327bbac5ec/examples/paths/main.go#L32)):*
<details><summary><i>Previously, nfarah86 (nfarah86) wrote…</i></summary><blockquote>

should we replace Println with something else? logger.Debug ? or use the zap library?  What are your thoughts?
</blockquote></details>

This is just an example program. It is intended to print to stdout.

---

*[pkg/paths/paths.go, line 44 at r1](https://reviewable.io/reviews/storj/storj/101#-LFT15Cat3UiU4RnVB9U-r1-44:-LFq29ZrEiI7ezZbU9bP:b60oje6) ([raw file](https://github.com/storj/storj/blob/74e6203d1e3bd701e53b299a4f9d98327bbac5ec/pkg/paths/paths.go#L44)):*
<details><summary><i>Previously, nfarah86 (nfarah86) wrote…</i></summary><blockquote>

may be I totally missed this, but what do you do with key?
</blockquote></details>

Here we derive the key that will be used to encrypt the next path segment in the next iteration of the for loop.

---


*Comments from [Reviewable](https://reviewable.io/reviews/storj/storj/101)*

**Comment by coyle** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @nfarah86)

---



*Comments from [Reviewable](https://reviewable.io/reviews/storj/storj/101#-:-LFwVFFB2QsDmtuHg1Fa:bnfp4nl)*

**Pull request merged**